src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -533,6 +533,15 @@ private function getArgumentsAsPhp(\DOMElement $node, $name, $file, $lowercase =
                         throw new InvalidArgumentException(sprintf('Tag "<%s>" with type="service_locator" only accepts maps of type="service" references in "%s".', $name, $file));
                     }
                     break;
+                case 'tagged_locator':
+                    if (!$arg->getAttribute('tag') && !$arg->getAttribute('index-by')) {
+                        throw new InvalidArgumentException(sprintf('Tag "<%s>" with type="tagged_locator" has no or empty "tag" or "index-by" attributes in "%s".', $name, $file));
+                    }
+
+                    $serviceLocatorArgument = new ServiceLocatorArgument([]);
+                    $serviceLocatorArgument->setTaggedIteratorArgument(new TaggedIteratorArgument($arg->getAttribute('tag'), $arg->getAttribute('index-by'), $arg->getAttribute('default-index-method') ?: null));
```

Comment:
what about removing the setter and accepting the TaggedIteratorArgument as argument instead, as 1st arg to the constructor?